tests/run/fused_types.pyx

```diff
@@ -510,3 +510,33 @@ def convert_to_ptr(cython.floating x):
         return handle_float(&x)
     elif cython.floating is double:
         return handle_double(&x)
+
+cdef double get_double():
+    return 1.0
+cdef float get_float():
+    return 0.0
+
+cdef call_func_pointer(cython.floating (*f)()):
+    return f()
+
+def test_fused_func_pointer():
+    """
+    >>> test_fused_func_pointer()
+    1.0
+    0.0
+    """
+    print(call_func_pointer(get_double))
+    print(call_func_pointer(get_float))
```

Comment:
Just wondering – does this also work for a … pointer to a function that takes a fused function as argument? Or something like that? I'd like to be sure that we don't start to lose relevant return types along the way somewhere when switching between "relevant" and "irrelevant" return types.

Reply:
Yes it does work - because the fused function argument is passed as a function pointer. I've added a test (that does to one level, because there's only so many levels of function pointers that I can think about)